### PR TITLE
Automate subscriber welcome email #766

### PR DESCRIPTION
**Pull request by duncantmiller**
Hi Joe, I'm submitting this as a draft PR because for some reason the tests hang when I run `bin/check`. Both `rails test` and `rails test:system` run fine. Also would you mind placing a test subscription and making sure everything runs correctly? I was able to do a test order but wasn't able to figure out if the webhook fired or not, so it would make me feel better if you tested it too.

Let me know if there is anything I can change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,8 +95,8 @@ stripe:
   private_key: sk_test_YOUR_TEST_STRIPE_KEY
   signing_secret: whsec_YOUR_SIGNING_SECRET
   price_ids:
-    part_time_plan: price_YOUR_PRODUCT_PRICE_ID
-    full_time_plan: price_ANOTHER_PRODUCT_PRICE_ID
+    part_time: price_YOUR_PRODUCT_PRICE_ID
+    full_time: price_ANOTHER_PRODUCT_PRICE_ID
```
Comment (joemasilotti) [inline README.md:99]: What's the reasoning behind this change?

File: app/mailers/business_mailer.rb [hunk 2]
```diff
@@ -12,6 +12,15 @@ def welcome
     )
   end
 
+  def subscribed
+    @business = params[:business]
+    from = Rails.configuration.emails.support_mailbox!
+    @developer_count = SignificantFigure.new(Developer.visible.count).rounded
+    subject = "Your RailsDevs subscription is live! 🎉 Let’s find your perfect match."
```
Comment (joemasilotti) [inline app/mailers/business_mailer.rb:19]: Let's move this to the English locale file.

File: app/views/business_mailer/subscribed.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<p><%= @business.first_name %>,</p>
```
Comment (joemasilotti) [inline app/views/business_mailer/subscribed.html.erb:1]:
```suggestion
<p>Hey <%= @business.first_name %>,</p>
```

File: app/views/business_mailer/subscribed.text.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<%= @business.first_name %>,
```
Comment (joemasilotti) [inline app/views/business_mailer/subscribed.text.erb:1]:
```suggestion
Hey <%= @business.first_name %>,
```

File: app/views/business_mailer/subscribed.html.erb (hunk 3 above)
Comment (joemasilotti) [reply app/views/business_mailer/subscribed.html.erb:1]: Also, let's remove the blank lines between `<p>` tags to match the other emails.

File: app/views/business_mailer/subscribed.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+<p>It’s Joe again, the founder of RailsDevs. Your subscription is live and you can message all <%= @developer_count %>+ Rails developers on the platform.</p>
```
Comment (joemasilotti) [inline app/views/business_mailer/subscribed.html.erb:2]:
```suggestion
<p>It’s Joe again, the founder of RailsDevs. Your subscription is live and you can now message all <%= @developer_count %>+ Rails developers on the platform.</p>
```

File: app/views/business_mailer/subscribed.text.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+It’s Joe again, the founder of RailsDevs. Your subscription is live and you can message all <%= @developer_count %>+ Rails developers on the platform.
+
+Get started by adding a filter or two when <%= link_to "browsing developer profiles", developers_url %>. Start with role level and work preference then try adding a specific term or technology.
+
+When you see a good fit click the Hire button. Your message goes right to their email inbox.
+
+Hit reply or <%= link_to "schedule a call", "https://savvycal.com/joemasilotti/RailsDevs" %> if you’d like help narrowing down your search!
```
Comment (joemasilotti) [inline app/views/business_mailer/subscribed.text.erb:9]: These won't render in a plain text email. Please move the link to the end of the line like in `BusinessMailer#welcome`.

File: docs/production_settings.md [hunk 7]
```diff
@@ -94,8 +94,8 @@ stripe:
   signing_secret:
   price_ids:
     legacy_plan:
-    part_time_plan:
-    full_time_plan:
+    part_time:
+    full_time:
```
Comment (joemasilotti) [inline docs/production_settings.md:98]: (same question as above)

File: test/mailers/business_mailer_test.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+class BusinessMailerTest < ActionMailer::TestCase
+  test "subscribed" do
+    BusinessMailer.with(business:).subscribed.deliver_later
+    assert_enqueued_emails 1
+  end
+
+  def business
+    businesses(:subscriber)
+  end
+end
```
Comment (joemasilotti) [inline test/mailers/business_mailer_test.rb:12]: We can delete this test. The `verify_mailer_previews` job ensures that none of them break.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
I tested a subscription and it's working! And `bin/check` seems to be working on my machine.

I made a few requests inline - once those are taken care of I will merge this.

File: README.md (hunk 1 above)
Comment (duncantmiller) [reply README.md:99]: We talked about this in person at the hackathon. When I tried using `part_time_plan` and `full_time_plan` under the stripe price_ids section of the development credentials file I was getting an error when trying to check out. When we switched it to `part_time` and `full_time` it worked and you remembered this had changed at some point and asked me to update the readme.

Comment (joemasilotti) [reply README.md:99]: Beautiful! I totally forgot about our conversation at the hackathon. Thank you for remembering it. :D

File: test/models/developer_test.rb [hunk 9]
```diff
@@ -213,7 +213,7 @@ class DeveloperTest < ActiveSupport::TestCase
     assert developer.featured?
     assert_includes Developer.featured, developer
 
-    travel 2.hours
+    travel 3.hours
```
Comment (duncantmiller) [inline test/models/developer_test.rb:216]: @joemasilotti not sure if this is the best solution here but this test was failing for me locally and in CI. I didn't change anything that would have affected it and when I reverted back to a commit that had been passing it also failed there, so I assume it has something to do with the time of day this is being run? I increased it an extra hour and now it passes.

Comment (joemasilotti) [reply test/models/developer_test.rb:216]: I'm assuming we are right on the edge case of an hour break in some way. Increasing the time to 3 hours is fine by me. I don't think it is worth diving into any more. Thanks for the quick fix!

File: README.md (hunk 1 above)
Comment (duncantmiller) [reply README.md:99]: 👍

File: app/views/business_mailer/subscribed.text.erb [hunk 10]
```diff
@@ -0,0 +1,15 @@
+Hey <%= @business.first_name %>,
+
+It’s Joe again, the founder of RailsDevs. Your subscription is live and you can message all <%= @developer_count %>+ Rails developers on the platform.
+
+Get started by adding a filter or two when browsing developer profiles <%= developers_url %>. Start with role level and work preference then try adding a specific term or technology.
```
Comment (joemasilotti) [inline app/views/business_mailer/subscribed.text.erb:5]:
```suggestion
Get started by adding a filter or two when browsing developer profiles. Start with role level and work preference then try adding a specific term or technology:  <%= developers_url %>
```

File: app/views/business_mailer/subscribed.text.erb [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+When you see a good fit click the Hire button. Your message goes right to their email inbox.
+
+Hit reply or schedule a call https://savvycal.com/joemasilotti/RailsDevs if you’d like help narrowing down your search!
```
Comment (joemasilotti) [inline app/views/business_mailer/subscribed.text.erb:9]:
```suggestion
Hit reply or schedule a call if you’d like help narrowing down your search: https://savvycal.com/joemasilotti/RailsDevs 
```

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Can you please add an entry to the CHANGELOG? I also requested two minor changes to the text version of the mailer.

Then we are good to merge!

**Comment by joemasilotti** [comment]
Thanks for your contribution, @duncantmiller! Merging in now :)

**Comment by duncantmiller** [comment]
Great glad to help!
